src/fundus/scraping/common_crawl/html.py

```diff
@@ -0,0 +1,66 @@
+from typing import Dict, Iterator, List, Optional, cast
+from urllib.parse import urlparse
+
+import requests
+from fastwarc import ArchiveIterator, WarcRecord, WarcRecordType
+from ftfy import guess_bytes
+
+from fundus.logging import basic_logger
+from fundus.publishers.base_objects import PublisherEnum
+from fundus.scraping.filter import URLFilter
+from fundus.scraping.html import HTML, WarcSource
+
+
+class CCNewsSource:
+    def __init__(self, *publishers: PublisherEnum, warc_path: str, headers: Optional[Dict[str, str]] = None):
+        self.publishers = publishers
+        self.warc_path = warc_path
+        self.headers = headers or {}
+
+        self._publisher_mapping: Dict[str, PublisherEnum] = {
+            urlparse(publisher.domain).netloc: publisher for publisher in publishers
+        }
+
+    def fetch(self, url_filter: Optional[URLFilter] = None) -> Iterator[HTML]:
+        domains = list(self._publisher_mapping)
+
+        def extract_body(record: WarcRecord) -> str:
+            raw_body = record.reader.read()
+            try:
+                return str(raw_body, encoding=record.http_charset)
+            except (UnicodeDecodeError, TypeError):
+                return guess_bytes(raw_body)[0]
```

Comment:
I ran into this as well and that's definitely on me for not sharing my thoughts.

According to [this](https://ftfy.readthedocs.io/en/v5.0.1/#a-note-on-encoding-detection) part of the `ftfy` documentation both come with their downside and I decided against `chardet` because of the single-byte part. In theory, this fallback mechanic shouldn't be used often, and I can't remember if it was ever needed, but we should log that case.

Reply:
> According to [this](https://ftfy.readthedocs.io/en/v5.0.1/#a-note-on-encoding-detection) part of the ftfy documentation both come with their downside and I decided against chardet because of the single-byte part.

Yes, unfortunately, that's how it is. Regarding the single vs multi-byte use case: Wouldn't multi-byte variants and language-specific codings be relevant for our library since it deals with multilingual text and different scripts e.g. Cyrillic? 

I did some testing with `ftfy` and `chardet`, running the crawler for ~60 seconds. All articles without a specified encoding were identified as UTF-8 by both `ftfy` and `chardet`. But I guess this is not representative since

- the number of articles, where the encoding mechanism failed was at ~0.1% and
- there were no special cases, where one or the other library's use case excels. (So no `sloppy-windows-1252` or `Latin-1` from ftfy's side. And no obscure language-specific encodings from chardet's side. But this is also to be expected as we do not crawl publishers with non-latin scripts, yet.)

Logs:
[ftfy.txt](https://github.com/flairNLP/fundus/files/14077042/ftfy.txt)
[chardet.txt](https://github.com/flairNLP/fundus/files/14077043/chardet.txt)

> In theory, this fallback mechanic shouldn't be used often, and I can't remember if it was ever needed, but we should log that case.

Yes, it's still ~0.1%.